qiskit_ibm_provider/ibm_backend_service.py

```diff
@@ -139,6 +139,8 @@ def backends(
         if instance:
             hgp = self._provider._get_hgp(instance=instance)
             backends = list(hgp.backends.values())
+            for backend in backends:
+                backend._instance = instance  # type: ignore
```

Comment:
Why not just pass the hgp when [initializing the IBMBackend instance](https://github.com/Qiskit/qiskit-ibm-provider/blob/10f1377df773a7f42a974891d28620b06f83d672/qiskit_ibm_provider/hub_group_project.py#L88-L92)? That way you are not changing a private attribute after the fact.